packages/lib/src/analytics/hooks/usePageVisibilityAnalytics.ts

```diff
@@ -0,0 +1,179 @@
+import { useEffect, useRef } from "react";
+import { useAnalyticsRoot } from "./useAnalyticsRoot";
+
+type PageVisibilityAction =
+  | {
+      name: "System Event page became visible";
+      "visibility.duration_hidden_ms": number;
+      "visibility.timestamp": string;
+    }
+  | {
+      name: "System Event page became hidden";
+      "visibility.duration_visible_ms": number;
+      "visibility.timestamp": string;
+    }
+  | {
+      name: "System Event page visibility session started";
+      "visibility.initial_state": "visible" | "hidden";
+      "visibility.timestamp": string;
+    }
+  | {
+      name: "System Event page visibility session ended";
+      "visibility.total_visible_ms": number;
+      "visibility.total_hidden_ms": number;
+      "visibility.state_changes": number;
+      "visibility.timestamp": string;
+    };
+
+interface UsePageVisibilityAnalyticsOptions {
+  /**
+   * The identifier for the page or component being tracked
+   */
+  identifier: string;
+  /**
+   * Additional attributes to include with every event
+   */
+  attributes?: Record<string, string | number | boolean>;
+  /**
+   * Whether to track the visibility changes
+   * @default true
+   */
+  enabled?: boolean;
+  /**
+   * Whether to track session start/end events
+   * @default true
+   */
+  trackSession?: boolean;
+  /**
+   * Minimum duration in milliseconds before tracking a visibility change
+   * Helps avoid tracking rapid tab switches
+   * @default 1000
+   */
+  minDuration?: number;
+}
+
+/**
+ * Hook that tracks page visibility changes and sends analytics events
+ * @param options Configuration options for the visibility tracking
+ * @param options.identifier The identifier for the page or component being tracked
+ * @param options.attributes Additional attributes to include with every event
+ * @param options.enabled Whether to track the visibility changes
+ * @param options.trackSession Whether to track session start/end events
+ * @param options.minDuration Minimum duration in milliseconds before tracking a visibility change
+ * @returns Object with current visibility state and manual tracking methods
+ * @example
+ * ```typescript
+ * const { isVisible } = usePageVisibilityAnalytics({
+ *   identifier: "Dashboard",
+ *   attributes: { "page.section": "overview" }
+ * });
+ * ```
+ */
+export const usePageVisibilityAnalytics = ({
```

Comment:
I think it would be preferable if we could use this hook without instantiating it on every page; it seems like it would be easy to forget to add this in the future. Do you think there's any way of doing that?

Also, leveraging the `route_name` param instead of the analytics object may be more helpful from a usability perspective. Is there any reason we must use `identifier`? At the very least, maybe omitting it would allow instantiating tracking in one parent component.